pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/RoundRobinScheduler.java

```diff
@@ -131,6 +138,20 @@ public int size() {
     return _ready.size() + _available.size();
   }
 
+  @Override
+  public void shutDown() {
+    if (_isShutDown) {
+      return;
+    }
+    while (!_ready.isEmpty()) {
+      _ready.poll().close();
+    }
+    while (!_available.isEmpty()) {
+      _available.poll()._opChain.close();
+    }
```

Comment:
should we add timeout for _opChain.close() ? is it possible opChain close to be stuck?

Reply:
See the comments above. Added a TODO for timeout consideration.